x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_item.tsx

```diff
@@ -257,8 +259,10 @@ export function WaterfallItem({
       isSelected={isSelected}
       hasToggle={hasToggle}
       onClick={(e: React.MouseEvent) => {
-        e.stopPropagation();
-        onClick(waterfallItemFlyoutTab);
+        if (onClick) {
+          e.stopPropagation();
+          onClick(waterfallItemFlyoutTab);
+        }
```

Comment:
Should we call `e.stopPropagation` regardless of the `onClick` state?

Reply:
This is actually intended. For embeddable waterfall`e.stopPropagation` we don't want to open the flyout when clicking on a node, but it's nice to collapse it. But on APM we do open the flyout, and in this case we don't want to propagate the click so the node is not collapsed.